lib/game/universe_test.go

```diff
@@ -0,0 +1,87 @@
+package game
+
+import (
+	"testing"
+)
+
+func TestUniverse(t *testing.T) {
+	t.Run("NewUniverse", func(t *testing.T) {
+		u := NewUniverse(24, 32)
+
+		if u.Height() != 24 {
+			t.Errorf("Expected height to be 24, got %d", u.Height())
+		}
+
+		if u.Width() != 32 {
+			t.Errorf("Expected width to be 32, got %d", u.Width())
+		}
+
+		var i uint32
+		for i = 0; i < u.Height()*u.Width(); i++ {
+			if u.Cell(i) != Dead {
+				t.Errorf("Expected cell %d to be dead, got %d", i, u.Cell(uint32(i)))
+			}
+		}
+	})
+
+	t.Run("AliveNeighbors", func(t *testing.T) {
+		u := NewUniverse(32, 32)
+
+		idx := u.GetIndex(12, 12)
+		u.cells[idx] = Alive
+		if u.Cell(idx) != Alive {
+			t.Errorf("Expected cell %d to be alive, got %d", idx, u.Cell(idx))
+		}
+
+		if u.AliveNeighbors(0, 0) != 0 {
+			t.Errorf("Expected cell %d to have 0 alive neighbors, got %d", 0, u.AliveNeighbors(0, 0))
+		}
+
+		if u.AliveNeighbors(11, 12) != 1 {
+			t.Errorf("Expected cell %d to have 1 alive neighbors, got %d", idx, u.AliveNeighbors(11, 12))
+		}
+
+		idx = u.GetIndex(10, 12)
+		u.cells[idx] = Alive
+
+		if u.AliveNeighbors(11, 12) != 2 {
+			t.Errorf("Expected cell %d to have 2 alive neighbors, got %d", idx, u.AliveNeighbors(11, 12))
+		}
+	})
+
+	t.Run("Tick", func(t *testing.T) {
+		u := NewUniverse(24, 32)
+		u.cells[u.GetIndex(10, 12)] = Alive
+		u.cells[u.GetIndex(11, 12)] = Alive
+		u.cells[u.GetIndex(12, 12)] = Alive
+		u.cells[u.GetIndex(11, 13)] = Alive
```

Comment:
It would probably make sense to add tests for all the four rules in the future 🤔